### PR TITLE
Feature | b - template

### PR DESCRIPTION
**Pull request by davidebukali**
Create template with alternative way to explore chapter contents #1001 

To activate `Show chapter contents` button, add links to respective pages using `add_content_links` field

### CODE REVIEW

**Review by edwinmp — CHANGES_REQUESTED** [review]

File: di_website/publications/models.py [hunk 1]
```diff
@@ -353,6 +353,12 @@ class Meta:
 
     colour = models.CharField(max_length=256, choices=COLOUR_CHOICES, default=RED)
 
+    read_online_button_text = models.CharField(max_length=256, default="Read Online", blank=True, null=True)
+
+    request_hard_copy_text = models.CharField(max_length=256, default="Request a hard copy", blank=True, null=True)
+
+    use_other_template = models.BooleanField(default=False)
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/publications/models.py]: Removed this field and using `get_template` function to point to the new template.

File: di_website/publications/models.py [hunk 2]
```diff
@@ -374,6 +380,9 @@ class Meta:
 
     content_panels = Page.content_panels + [
         FieldPanel('colour'),
+        FieldPanel('read_online_button_text'),
+        FieldPanel('request_hard_copy_text'),
+        FieldPanel('use_other_template'),
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/publications/models.py]: Updated to use a `MultiFieldPanel`

File: di_website/publications/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/publications/models.py]: Moved to publications mixin

File: di_website/templates/publications/chapter_content.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            {% if block.block_type == 'links' %}
+                {% if forloop.last %}
+                    <li class="list-item space-section-bottom--small" data-js="toggleable-list--item"><a href="{{block.value.link}}">{{block.value.title}}</a></li>
+                {% else %}
+                    <li class="list-item" data-js="toggleable-list--item"><a href="{{block.value.link}}">{{block.value.title}}</a></li>
+                {% endif %}    
+            {% endif %}
+        {% endfor %}
+    </ul>
+{% endif %}
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/templates/publications/chapter_content.html]: Updated

File: di_website/templates/includes/partials/call-to-action-inline.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {% load wagtailcore_tags %}
 
-<div class="cta">
+<div class="cta" id="hardcopy">
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/templates/includes/partials/call-to-action-inline.html]: Updated

File: di_website/templates/includes/heroes/hero-pubs.html [hunk 5]
```diff
@@ -44,9 +44,16 @@ <h2 class="hero__subtitle">{{ page.hero_text|content }}</h2>
       {% if page.hero_link and page.hero_link_caption %}
         <a href="{% pageurl page.hero_link %}" class="button button--reverse">{{ page.hero_link_caption }}</a>
       {% endif %}
+
+      {% if page.read_online_button_text %}
```
Comment (edwinmp) [inline di_website/templates/includes/heroes/hero-pubs.html:48]: You named this `read_online_text` ... clearly forgot to test it or you'd have noticed the button doesn't show ... make the change in the model as this is the better name.

**Review by edwinmp — CHANGES_REQUESTED** [review]
Sorry this took a while... power went off while I was testing ...

File: di_website/publications/mixins.py [hunk 6]
```diff
@@ -176,3 +177,11 @@ class Meta:
...
     def call_to_action(self):
         return self.get_parent().specific.publication_cta.filter(inherit=True)
+
+
+class HeroButtonMixin(models.Model):
+    read_online_text = models.CharField(max_length=256, default="Read Online", blank=True, null=True)
+    request_hard_copy_text = models.CharField(max_length=256, default="Request a hard copy", blank=True, null=True)
+
+    class Meta:
+        abstract = True
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/publications/mixins.py]: Updated

File: di_website/templates/includes/heroes/hero-pubs.html [hunk 7]
```diff
@@ -44,9 +44,16 @@ <h2 class="hero__subtitle">{{ page.hero_text|content }}</h2>
...
+        <a href="#exec-summary" class="button button--reverse">{{ page.read_online_button_text }}</a>
+      {% endif %}
       {% if page.publication_downloads_list|length > 0 or page.data_downloads_list|length > 0 %}
         <a href="#downloads" class="button button--reverse">Downloads</a>
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/templates/includes/heroes/hero-pubs.html]: Updated

File: di_website/templates/includes/heroes/hero-pubs.html [hunk 8]
```diff
@@ -44,9 +44,16 @@ <h2 class="hero__subtitle">{{ page.hero_text|content }}</h2>
...
       {% endif %}
+      {% if page.request_hard_copy_text %}
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/templates/includes/heroes/hero-pubs.html]: Added `@cached_property` that checks for top CTAs

File: di_website/templates/includes/heroes/hero-pubs.html (hunk 5 above)
Comment (davidebukali) [reply di_website/templates/includes/heroes/hero-pubs.html:48]: Updated model

File: di_website/templates/includes/partials/call-to-action-inline.html [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {% load wagtailcore_tags %}
 
-<div class="cta">
+<div class="cta" {% if item.position == 'top' and forloop.first %} id="hardcopy" {% endif %}>
```
[the remark this answers is not in the document]
Comment (davidebukali) [reply di_website/templates/includes/partials/call-to-action-inline.html]: Removed unnecessary check

**Review by edwinmp — APPROVED** [review]

**Pull request merged**